### DISCUSSION TITLE
Prettify Typescript Error

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [zed-industries/zed#24607] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

In editors like the JetBrains (built-in) ones or VSCode ([through a plugin](https://github.com/yoavbls/pretty-ts-errors)), we can view our TS errors way prettier than they normally are. If you are working with complex type systems and you get a type error, most of the time, this error does not help at all, not because the error is bad but because its styling is ugly.

### Environment

I don't think that matters but here:

Zed: v0.160.7 (Zed)
OS: macOS 15.1.0
Memory: 8 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help present your vision of the feature

<img height="400" src="https://github.com/user-attachments/assets/33513a11-10ad-4b84-95bd-7fd8a1bb081c">

And that:
<img height="400" src="https://github.com/user-attachments/assets/788c82a6-7bb7-405e-a9b8-bd3bbb8528dc">

Instead of that:
<img width="400" alt="image" src="https://github.com/user-attachments/assets/80541be1-9b67-42fa-a9b2-a972e880e6fc">

**Comment by radiosilence** [comment]
This would be awesome - too often am I faced with a literal wall of characters with no real way of picking out what exactly the f is broken about some data.

**Comment by TheBit** [comment]
Any plans for this?
